ddl/ddl_api.go

```diff
@@ -2167,18 +2173,31 @@ func (d *ddl) RecoverTable(ctx sessionctx.Context, recoverInfo *RecoverInfo) (er
 		return infoschema.ErrTableExists.GenWithStackByArgs(tbInfo.Name)
 	}
 
-	tbInfo.State = model.StateNone
+	recoverInfo.TableInfo.State = model.StateNone
```

Comment:
Why do you change it?

Reply:
ye~ look like this is useless

Reply:
Look like it was a refactor and keep the behavior same as older.